karoo_gp/branch.py

```diff
@@ -230,9 +231,18 @@ def display(self, *args, method='viz', **kwargs):
             return self.display_viz(*args, **kwargs)
 
     def display_list(self, prefix=''):
-        output = prefix + repr(self.node) + '\n'
+        _type = 'term' if type(self.node) is Terminal else 'func'
+        _symbol = self.node.symbol
+        _parent = '' if self.parent is None else self.parent.id
+        _arity = 0 if _type == 'term' else self.node.arity
+        _children = '' if not self.children else [c.id for c in self.children]
```

Comment:
```suggestion
        _children = [] if not self.children else [c.id for c in self.children]
```
The type should be consistent, either always a `list` or always a `str`.  Otherwise you could also set it to `None`, but that might cause other problems.